lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -1172,7 +1172,7 @@ void TypeChecker::notePlaceholderReplacementTypes(Type writtenType,
       }
 
       if (auto *origRepr =
-              placeholder->getOriginator().dyn_cast<PlaceholderTypeRepr *>()) {
+              placeholder->getOriginator().dyn_cast<TypeRepr *>()) {
```

Comment:
We should add an assert here that that makes sure that `origRepr` was ` PlaceholderTypeRepr`.